perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -518,6 +593,148 @@ def SingleStreamThroughputBenchmark(storage_schema, host_to_connect=None):
                   host_to_connect=host_to_connect)
 
 
+def MultiStreamThroughputBenchmark(storage_schema, host_to_connect=None):
+  """A benchmark test for single stream upload and download throughput.
+
+  Args:
+    storage_schema: The schema of the storage provider to use, e.g., "gs"
+    host_to_connect: An optional host endpoint to connect to.
+
+  The object size is given by FLAGS.object_size_kb. The total number
+  of objects to transfer (summed across all streams) is given by
+  FLAGS.iterations. The benchmark starts at FLAGS.write_start_time, if
+  given, or else immediately. Reads start at FLAGS.read_start_time, if
+  given, or immediately after the writes end.
+
+  Raises:
+    LowAvailabilityError: when the storage provider has failed a high number of
+        our RW requests that exceeds a threshold (>5%), we raise this error
+        instead of collecting performance numbers from this run.
+  """
+
+  result_queue = Queue.Queue()
+
+  logging.info('About to create processes for multistream.')
+
+  processes = [threading.Thread(target=MultiStreamThroughputWorker,
+                                args=(storage_schema,
+                                      host_to_connect,
+                                      FLAGS.object_size_kb,
+                                      FLAGS.objects_per_stream,
+                                      FLAGS.write_start_time,
```

Comment:
For write and read start time, I'd like to suggest that these should not be plumbed into the thread workers. The workers should be made as dumb as possible and all they do is: write X number of objects or read X number of objects.

The intelligence of when to start write or read should be contained in the caller, that is, here.

Regarding the write_start_time and read_start_time design:

write-start-time is easy to implement: just start the writer threads when current time > write_start_time here (or start immediately if write_start_time is none).

Read_start_time would be slightly different: it's the responsibility of the caller here to wait for all writer threads to finish writing, and then it would launch the reader threads if current time > read_start_time.  this way, we will always get a very clean separation between the write phase and read phase among these threads on each VM.

Yes, above implies that we should split the thread worker into writer and reader so we have more flexibility in controlling them, and have better separation.